spec/lib/whitehall_importer/embed_body_references_spec.rb

```diff
@@ -12,49 +12,106 @@
       expect(govspeak_body).to eq("[Contact:#{content_id}]")
     end
 
-    it "converts Whitehall image syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
+    it "converts Whitehall image bang embeds to govspeak Image embeds" do
+      body = described_class.call(body: "!!1", images: ["file.jpg"])
+
+      expect(body).to eq("[Image:file.jpg]")
+    end
+
+    it "converts Whitehall attachment bang embeds to govspeak Attachment embeds" do
+      body = described_class.call(body: "!@1", attachments: ["file.pdf"])
+
+      expect(body).to eq("[Attachment:file.pdf]")
+    end
+
+    it "ignore Whitehall image bang embeds that are neither start of the string or preceeded by new lines" do
+      body = described_class.call(
         body: "!!1 test !!2",
-        images: ["foo.png", "bar.jpg"],
+        images: ["file.png", "file.jpg"],
       )
 
-      expect(govspeak_body).to eq("[Image:foo.png] test [Image:bar.jpg]")
+      expect(body).to eq("[Image:file.png] test !!2")
     end
 
-    it "removes any image markdown that doesn't resolve to an image" do
-      govspeak_body = described_class.call(
-        body: "Bar !!2 Baz",
-        images: ["foo.png"],
+    it "ignore Whitehall attachment bang embeds that are neither start of the string or preceeded by new lines" do
+      body = described_class.call(
+        body: "!@1 test !@2",
+        attachments: ["file.pdf", "file.csv"],
       )
 
-      expect(govspeak_body).to eq("Bar  Baz")
+      expect(body).to eq("[Attachment:file.pdf] test !@2")
     end
 
-    it "converts Whitehall attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!@1 test !@2",
-        attachments: ["file.pdf", "download.csv"],
+    it "ignore any image markdown that doesn't resolve to an image" do
```

Comment:
I don't think this text is correct as I believe this didn't resolve because this has no new lines before it rather than because the image doesn't resolve. 

If the image doesn't resolve I thought we ended up with an empty string?

Reply:
Good spot - thanks.  Have update the test to clarify the intent.